### DISCUSSION TITLE
rounded-full prevents other rounding sizes from functioning on element

### DISCUSSION THREAD

**Discussion by zolrath** [tailwindlabs/tailwindcss#1726] [Help]
Hello!
When attempting to create a button that is `rounded-full`  but with a single, less round corner via `rounded-br-lg` I am not seeing the behavior I'd expect given the rest of the composability of tailwind.

`rounded-full` sets `border-radius: 9999px` and it seems the only way to approximate `rounded-br-lg` was to add `border-bottom-right-radius: 3500px`

![border-full](https://user-images.githubusercontent.com/454563/81225331-eb631c80-8f9d-11ea-92c8-6feff42da070.png)

**Comment by estevanmaito** [comment]
As per the CSS Border spec https://www.w3.org/TR/css-backgrounds-3/#corner-overlap

> Corner curves must not overlap: When the sum of any two adjacent border radii exceeds the size of the border box, UAs must proportionally reduce the used values of all border radii until none of them overlap

That's why you needed to add 3500px to start seeing something: _proportionally reduce_

So the only way to preciselly give `border-radius` to an element, like your case, with different radii, is to do it manually.

An example: for this button with `.h-10` (`height: 2.5rem`), the maximum allowed border size, until they start to overlap is 1/2, or `1.25rem`.

```html
<button class="h-10 px-6 bg-indigo-200 text-indigo-900 btn">Lorem ipsum</button>
```

```css
.btn {
  border-radius: 1.25rem 1.25rem .5rem 1.25rem;
}
```

![Screenshot_2020-05-17 GRpYbep(1)](https://user-images.githubusercontent.com/8527573/82156574-34df2180-9852-11ea-95cf-102f256087cf.png)

I don't know if would be possible for Tailwind to provide different rounded border utilities given the amount of sizes an element could have (taking into account height, padding, line-height, etc.)
